eido/conversion.py

```diff
@@ -76,41 +76,36 @@ def run_filter(prj, filter_name, verbose=True, plugin_kwargs=None):
     func = installed_plugins[filter_name]
 
     # run filter
-    conv_results = func(prj, **plugin_kwargs)
+    conv_result = func(prj, **plugin_kwargs)
 
     # if paths supplied, write to disk
     if paths is not None:
-        # make out dirs if they don't already exist
-        for path in paths:
-            if not os.path.exists(paths[path]):
-                os.makedirs(os.path.dirname(paths[path]), exist_ok=True)
-
-        # extract potential results
-        project_result = conv_results.get("project")
-        samples_result = conv_results.get("samples")
-
-        # write config conversion result
-        if project_result is not None:
-            with open(paths["cfg"], "w") as f:
-                f.write(project_result)
-
-        # write samples conversion result
-        if samples_result is not None:
-            with open(paths["samples"], "w") as f:
-                f.write(samples_result)
+        # map conversion result to the
+        # specified path
+        for result_key in conv_result:
```

Comment:
Can you use something like:

```
for result_key, result_path in conv_result.items(): 
...
```

Reply:
it saves a line of code :)

Reply:
sure

Reply:
Oh. `result_path` is acquired using `paths.get(result_key)` such that if it doesn't exist in `paths` it is going to be `None`.

I believe that this:
```
for result_key, result_path in conv_result.items(): 
...
```

Reply:
Thoughts on this?

Reply:
yes I think you're right.

Reply:
my bad!